### PR TITLE
[W5.4][T10-A2]YapZuoXuan

### PR DESCRIPTION
**Pull request by wishingmaid**
added new test cases and new conditions for displaying addedHistory

### CODE REVIEW

**Comment by Infinity-Ace** [comment]
It's nice to know your history of what you did, but eventually the history gets too long and isn't very comfortable for users. However, solid implementation!

**Comment by qihao27** [comment]
A convenient method to recall deleted contacts if recovery is required.

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -81,13 +83,25 @@ private void exit() {
...
         Command command;
+        final ArrayList<CommandResult> recentAdded = new ArrayList<CommandResult>();
         do {
             String userCommandText = ui.getUserCommand();
             command = new Parser().parseCommand(userCommandText);
             CommandResult result = executeCommand(command);
             recordResult(result);
             ui.showResultToUser(result);
 
+
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/Main.java]: Unnecessary empty lines.

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -81,13 +83,25 @@ private void exit() {
...
+
+            if(command instanceof AddCommand && !result.feedbackToUser.equals(AddCommand.MESSAGE_DUPLICATE_PERSON)) {
+                recentAdded.add(0,result);
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/Main.java]: magic number and no space after a comma.

File: src/seedu/addressbook/Main.java [hunk 3]
```diff
@@ -81,13 +83,25 @@ private void exit() {
...
         do {
             String userCommandText = ui.getUserCommand();
             command = new Parser().parseCommand(userCommandText);
             CommandResult result = executeCommand(command);
             recordResult(result);
             ui.showResultToUser(result);
 
+
+
+            if(command instanceof AddCommand && !result.feedbackToUser.equals(AddCommand.MESSAGE_DUPLICATE_PERSON)) {
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/Main.java]: No white space after `if`.

File: src/seedu/addressbook/Main.java [hunk 4]
```diff
@@ -81,13 +83,25 @@ private void exit() {
...
+                recentAdded.add(0,result);
+                System.out.println("History of recently added:");
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/Main.java]: Shouldn't it be in TextUI class?

File: src/seedu/addressbook/Main.java [hunk 5]
```diff
@@ -81,13 +83,25 @@ private void exit() {
...
+                for (CommandResult addHistory : recentAdded) {
+                    System.out.println(addHistory.feedbackToUser);
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/Main.java]: Shouldn't it be in TextUI class?

**Comment by jeffryhartanto** [comment]
@wishingmaid Some comments added. Please close the PR after reading comments.

Good job for adding a new feature, updating the user guide and I/O test and passing Travis check. Since the additional feature doesn't require you to write a JUnit test, you can still obtain the 5 participation marks. However, I advise you to practice writing JUnit tests.

**Pull request closed without merging**